### PR TITLE
[TypeDeclaration] Handle nullable MockObject on TypedPropertyFromCreateMockAssignRector

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://github.com/rectorphp/rector-src/pull/6178#pullrequestreview-2195119530

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

File: rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php [hunk 1]
```diff
@@ -100,17 +106,35 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
-            $type = $this->trustedClassMethodPropertyTypeInferer->inferProperty(
-                $node,
+            if (! $classReflection instanceof ClassReflection) {
+                $classReflection = $this->reflectionResolver->resolveClassReflection($node);
+            }
```
Comment (TomasVotruba) [inline rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php:111]: This should not be needed.

File: rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/nullable_mock_object.php.inc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    protected function setUp(): void
+    {
+        $this->someMock = $this->createMock(SomeMockedClass::class);
+    }
+
+    protected function tearDown(): void
+    {
+        $this->someMock = null;
+    }
```
Comment (TomasVotruba) [inline rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/nullable_mock_object.php.inc:44]: Could you also add test fixture with property set in another method except 'setUp()`?

File: rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php (hunk 1 above)
Comment (samsonasik) [reply rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php:111]: it initialized as null before loop, I don't want to get class reflection when class doesn't have property, so early instanceof is used for tweak, then early skip when it got null

File: rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/nullable_mock_object.php.inc (hunk 2 above)
Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/nullable_mock_object.php.inc:44]: do you want to skip if it never assigned in `setUp`, but assigned in another method?

Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/nullable_mock_object.php.inc:44]: see https://github.com/rectorphp/rector-src/pull/6180

File: rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php (hunk 1 above)
Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php:111]: This rule should be simple as possible. Please remove the `$classReflection = $this->reflectionResolver->resolveClassReflection($node);` and use simple check as before.

Also, wait for review before next merges. I need to review those first

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector.php:111]: sure, see https://github.com/rectorphp/rector-src/pull/6181

**Pull request merged**